src/errors.rs

```diff
@@ -106,3 +116,10 @@ pub enum IncomingMsgErr {
     /// Unsupported version
     UnsupportedVersion,
 }
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+/// Security Error
+pub enum SecurityError {
+    /// Too many orphan nonces; probably due to an attack
+    TooManyOrphanNonces,
```

Comment:
The nonce is an invisible notion for the user, it must not appear in the error names. Moreover, this error is not necessarily a security problem, it can be caused by a connection of very bad quality.